tests/test_immutableset.py

```diff
@@ -222,6 +222,28 @@ def test_singleton_index(self):
         with self.assertRaises(IndexError):
             s[-25]
 
+    def test_singleton_slice(self):
+        s = ImmutableSet.of([1])
+
+        self.assertEqual(s, s[0:1])
+        self.assertEqual(s, s[-1:1])
+
+        self.assertEqual(ImmutableSet.empty(), s[1:5])
+        self.assertEqual(ImmutableSet.empty(), s[51:56])
+        self.assertEqual(ImmutableSet.empty(), s[-5:-1])
+        self.assertEqual(ImmutableSet.empty(), s[-1:0])
+        self.assertEqual(ImmutableSet.empty(), s[-2:0])
```

Comment:
Maybe leave off the zero of this one so we know the default case works?
```suggestion
        self.assertEqual(ImmutableSet.empty(), s[-2:])
```